influxdb3/src/commands/serve.rs

```diff
@@ -167,6 +167,16 @@ pub struct Config {
         action
     )]
     pub query_log_size: usize,
+
+    // TODO - make this default to 70% of available memory:
+    /// The size limit of the open segments in the write buffer.
```

Comment:
I think this would require us doing clap by hand instead of derive. I don't think you can pass in a function to call that can calculate this dynamically at least not with derive.

Reply:
You can get away with this using a newtype with custom `FromStr` impl. We have one or two examples of this in the load generator, e.g.,
https://github.com/influxdata/influxdb/blob/5cb7874b2ca7681588592b92e419a3aaabcca11b/influxdb3_load_generator/src/commands/common.rs#L83-L111

Reply:
I think doing this in a followup is good. I know there were some tricks to finding the available memory and some PR in IOx that just had to do something to make sure it worked in specific environments.